frontend/src/redux/reducers/ui.tsx

```diff
@@ -42,6 +43,10 @@ export interface UIState {
       headerActions: {
         [name: string]: HeaderActionFunc;
       };
+      pluginAppendedDetailViews: Array<{
+        sectionName: string;
+        sectionFunc: () => { title: string; component: (props: any) => JSX.Element | null };
```

Comment:
This should be `(resource: KubeObject) => { title?: string; component?: (props: any) => JSX.Element | null } | null | undefined;`

title can be optional if we can take the sectionName. Otherwise, let's keep it mandatory as you had.
But the whole result can be null | undefined, because if you want not to have anything showing, you maybe just return null/undefined...